plugin-dev/Source/Sentry/Private/Android/SentrySubsystemAndroid.cpp

```diff
@@ -169,6 +169,32 @@ USentryId* SentrySubsystemAndroid::CaptureEventWithScope(USentryEvent* event, co
 	return SentryConvertorsAndroid::SentryIdToUnreal(*id);
 }
 
+USentryId* SentrySubsystemAndroid::CaptureException(const FString& type, const FString& message, int32 framesToSkip)
+{
+	// add marker tags specific for Unreal assertions
+	SetTag(TEXT("sentry_unreal_exception"), TEXT("assert"));
+	SetTag(TEXT("sentry_unreal_exception_skip_frames"), FString::Printf(TEXT("%d"), framesToSkip));
+	SetTag(TEXT("sentry_unreal_exception_type"), type);
+	SetTag(TEXT("sentry_unreal_exception_message"), message);
+
+	PLATFORM_BREAK();
+
+	return nullptr;
+}
+
+USentryId* SentrySubsystemAndroid::CaptureAssertion(const FString& type, const FString& message)
+{
+	return CaptureException(type, message, 8);
+}
+
+USentryId* SentrySubsystemAndroid::CaptureEnsure(const FString& type, const FString& message)
+{
+	auto id = FSentryJavaObjectWrapper::CallStaticObjectMethod<jobject>(SentryJavaClasses::SentryBridgeJava, "captureException", "(Ljava/lang/String;Ljava/lang/String;)Lio/sentry/protocol/SentryId;",
```

Comment:
I don't follow. Why can we call `CaptureException` from in `CaptureAssertion` but we have to go straight to the bridge?

Reply:
That's a good point. I've moved the assertion-specific logic from `CaptureException` to `CaptureAssertion`. Also, in this case we can leave the `CaptureException` interface stub empty since on Android assert/ensure handling flow is different from other platforms.